### ISSUE TITLE
Some questionns on test vectors and sample algorithm

### ISSUE THREAD

**Issue by HonDuan** [mitre/FMACM#4]
1. What is the expected value in PSI/MSI step change in FimTv03?
   To our understanding, the step change in PSI/MSI is generated by the jump of target aircraft's position. But we only observed 1 second (time spacing) step change in PSI/MSI because some filter is implemented in the ASSAP the position jump is smoothed. It seems 1 second step is too small to make the commanded speed change. What is the expected change in PSI/MSI in this test?
2. Limitation of speed profile
   In the MOPS requirement, FRAC.076, it says ASSAP shall (FRAC.076) limit the IM Speed to be within 15% of Ownship’s Airspeed Profile. But in the sample algorithm, the commanded speed is boundary by +/-10% of precalculated speed. Should it be 15% rather than 10%?
   Do we need to implement this airspeed limitation when using MACH as well? 
3. PSI/MSI calculation.
   Is there any implementation provided in sample algorithm to calculate PSI/MSI? Or we can only develop by ourselves according to the definition in the MOPS?

**Comment by sbowman-mitre** [comment]
The first question will need more time to answer than I currently have. I think I can answer 2 & 3 quickly. 

2) Yes,  the requirement changed from 10% to 15% during the most recent MOPS meeting in June. That was after the code had been released to RTCA. So, yes, your code should use 15% (or whatever value is indicated in the most recent version of the MOPS). And, yes, it should be applied to Mach also.

3) In the code provided here on github, these concepts are not fully implemented. However, in the Sample Algorithm code provided to RTCA, there is code to calculate PSI & MSI. Unfortunately, it's not as simple as looking at one method (because the code was developed at the same time that the overall concept was still being defined in committee). I recommend you examine getPathLengthFromPos() in Aircraft_Calculations.cpp. This is at the core of calculating the spacing interval. Also, look at IM_Time_To_Go_KinematicTM.cpp to see how getPathLengthFromPos() is used to calculate current_distance, true_distance, and target_distance. 
Note also that there is a slight difference in how MSI is calculated depending on the type of operation. If time-based, use one way. If distance-based, use another. That should be clearly seen in the MOPS.

Does that help?

**Comment by sbowman-mitre** [comment]
Hi @HonDuan , I'm answering 1) now. The change was intended to model a 0.45 g change in speed over 1 second. So, there should be about 10 seconds of coasting which causes linear extrapolation. Then, the next received ADS-B state should cause the 0.45 g step change in the PSI/MSI relative to the extrapolated data. Is this consistent with what you are seeing?

**Comment by sbowman-mitre** [comment]
Have your questions been answered for this issue?

**Comment by HonDuan** [comment]
Maybe I see the step change in the raw data(ADS-b raw squitter) but cannot see it in the position report because some filter implemented in the tracking. 
Anyway I try to re-run the TV03 and send you the result.

**Comment by whtoyou** [comment]
In Do-361, "The test vectors, described in APPENDIX G, are provided in the form of a zip
archive." and "The FIM Sample Algorithm source code is provided in the form of a zip archive".
Where can find the zip archive?

**Comment by sbowman-mitre** [comment]
Hi @whtoyou , the zip archive is only available from [RTCA](https://www.rtca.org/content/list-available-documents). You will need to contact them to acquire the files.

But before you go through that effort, let me clarify what you'll get. The code you'll receive is not a fully functional library. It is not guaranteed to compile and some implementation details have been removed to protect MITRE's intellectual property. It basically just sketches out the way we implemented it as an example. I hope that helps.
